akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala

```diff
@@ -314,15 +333,33 @@ private[projection] class R2dbcOffsetStore(
   }
 
   private def readTimestampOffset(): Future[Option[TimestampOffset]] = {
-    idle.set(false)
+    implicit val sys = system // for implicit stream materializer
+    triggerDeletionPerSlice.clear()
     val oldState = state.get()
 
-    dao.readTimestampOffset().map { recordsWithKey =>
+    val offsetSliceReadParallelism = 10 // FIXME config
+
+    val recordsWithKeyFut =
+      Source(minSlice to maxSlice)
+        .mapAsyncUnordered(offsetSliceReadParallelism) { slice =>
+          dao.readTimestampOffset(slice)
+        }
+        .mapConcat(identity)
+        .runWith(Sink.seq)
+        .map(_.toVector)(ExecutionContext.parasitic)
+
+    recordsWithKeyFut.map { recordsWithKey =>
       clearInflight()
       clearForeignOffsets()
       clearLatestSeen()
 
-      val newState = State(recordsWithKey.map(_.record))
+      val newState = {
+        val s = State(recordsWithKey.map(_.record))
+        // FIXME shall we evict here, or how does that impact the logic for moreThanOneProjectionKey and foreignOffsets?
```

Comment:
?

Reply:
I think that should be ok, as that starting offset logic is based on latest by slice already.